### DISCUSSION TITLE
Presence detector

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7061] [Ideas and suggestions]
"Presence detector" as a separate detector or as advanced motion "motion detector" settings.
For example
We have a "motion detector" but no "presence detector". When there is no movement, the motion detector shuts down. A separate sensor (or additional function in the motion sensor) is needed to transmit the signal continuously while the object is in the detection zone. In this way many ideas can be fully realized. When I use a motion detector, my guns moved into the submarine body as they approach the walls, but they come back out if there’s no movement. A presence detector would solve the problem.

**Comment by Ek-Videogames** [comment]
### Setting the speed requirement to 0 makes a motion sensor act as a constant for your purposes.

However - Motion sensors do have some other bugs though when used outside the submarine-hull which can cause them to stop working.
